src/measure_theory/vector_measure.lean

```diff
@@ -0,0 +1,617 @@
+    { exact h i hi },
+    { simp_rw [measure_of_not_measurable_set _ hi] } }
+end
+
+@[ext] lemma ext {s t : vector_measure α M}
+  (h : ∀ i : set α, measurable_set i → s i = t i) : s = t :=
+(ext_iff s t).2 h
+
+variables [t2_space M] {v : vector_measure α M} {s : signed_measure α} {f : ℕ → set α}
+
+lemma measure_Union_has_sum [encodable β] {f : β → set α}
+  (hf₁ : ∀ i, measurable_set (f i)) (hf₂ : pairwise (disjoint on f)) :
+has_sum (λ i, v (f i)) (v (⋃ i, f i)) :=
+begin
+  set g := λ i : ℕ, ⋃ (b : β) (H : b ∈ encodable.decode₂ β i), f b with hg,
+  have hg₁ : ∀ i, measurable_set (g i),
+  { exact λ _, measurable_set.Union (λ b, measurable_set.Union_Prop $ λ _, hf₁ b) },
+  have hg₂ : pairwise (disjoint on g),
+  { exact encodable.Union_decode₂_disjoint_on hf₂ },
+  have := v.measure_of_disjoint_Union hg₁ hg₂,
+  rw [hg, encodable.Union_decode₂] at this,
+
+  have hg₃ : (λ (i : β), v (f i)) = (λ i, v (g (encodable.encode i))),
+  { ext, rw hg, simp only,
+    congr, ext y, simp only [exists_prop, mem_Union, option.mem_def],
+    split,
+    { intro hy,
+      refine ⟨x, (encodable.decode₂_is_partial_inv _ _).2 rfl, hy⟩ },
+    { rintro ⟨b, hb₁, hb₂⟩,
+      rw (encodable.decode₂_is_partial_inv _ _) at hb₁,
+      rwa ← encodable.encode_injective hb₁ } },
+
+  rw [summable.has_sum_iff, this, ← tsum_Union_decode₂],
+  { exact v.measure_of_empty },
+  { rw hg₃, change summable ((λ i, v (g i)) ∘ encodable.encode),
+    rw function.injective.summable_iff encodable.encode_injective,
+    { exact (v.measure_of_disjoint_Union_has_sum hg₁ hg₂).summable },
+    { intros x hx,
+      convert v.measure_of_empty,
+      simp only [Union_eq_empty, option.mem_def, not_exists, mem_range] at ⊢ hx,
+      intros i hi,
+      exact false.elim ((hx i) ((encodable.decode₂_is_partial_inv _ _).1 hi)) } }
+end
+
+lemma measure_Union [encodable β] {f : β → set α}
+  (hf₁ : ∀ i, measurable_set (f i)) (hf₂ : pairwise (disjoint on f)) :
+v (⋃ i, f i) = ∑' i, v (f i) :=
+(measure_Union_has_sum hf₁ hf₂).tsum_eq.symm
+
+lemma measure_of_union {A B : set α}
+  (h : disjoint A B) (hA : measurable_set A) (hB : measurable_set B) :
+  v (A ∪ B) = v A + v B :=
+begin
+  rw [union_eq_Union, measure_Union, tsum_fintype, fintype.sum_bool, cond, cond],
+  exacts [λ b, bool.cases_on b hB hA, pairwise_disjoint_on_bool.2 h]
+end
+
+lemma measure_of_diff {A B : set α} (hA : measurable_set A) (hB : measurable_set B)
+  (h : A ⊆ B) : v A + v (B \ A) = v B :=
+begin
+  rw [← measure_of_union disjoint_diff hA (hB.diff hA), union_diff_cancel h],
+  apply_instance,
+end
+
+lemma measure_of_diff' {M : Type*} [add_comm_group M]
+  [topological_space M] [t2_space M] {v : vector_measure α M}
+  {A B : set α} (hA : measurable_set A) (hB : measurable_set B)
+  (h : A ⊆ B) : v (B \ A) = v B - (v A) :=
+begin
+  rw [← measure_of_diff hA hB h, add_sub_cancel'],
+  apply_instance,
+end
+
+end vector_measure
+
+/-- An alternative definition of `signed_measure` which does not require the
```

Comment:
I have to admit I am not really convinced by this `signed_measure'` business, because I can't think of a single nonartifical example where the condition for `signed_measure'` would be easier to check than the one for `signed_measure`. What bothers me is that you're duplicating the API while in the end they're exactly the same objects. We have this kind of things in some places in mathlib (for instance an alternate constructor for inner product spaces), but only for situations where the assumptions for the other constructor are genuinely easier to check than for the main one. Do you have an example in mind where this makes a difference? Otherwise, I'd probably just drop the `signed_measure'` alltogether (at least from this PR, to make it easier to review).

Reply:
I used it in `of_measure` though I would think I will be able to prove it without `signed_measure'`. I can give that a go but I suspect the proof might be very long since the proof is the same as the one for `signed_measure'`

Reply:
But I see what you mean. I think there is probably no other use for `signed_measure'` than `of_measure` and so we can just move the proofs out and prove it that way

Reply:
I've now removed `signed_measure'` and simply proved summability for `of_measure` instead